### ISSUE TITLE
Bug: Wrong installation permissions on Mac High Sierra

### ISSUE THREAD

**Issue by gasserles** [dbeaver/dbeaver#3485]
For safety/security, I have two accounts set up on my High Sierra
(10.3.4) Macbook Pro: one as an administrator with "sudo" priveleges,
and one as a regular user with no "sudo" priveleges. These have
separate home directories of course.

Using the Mac administrator account, I downloaded the most recent
Community Edition 5.0.5..dmg installer for the Mac:
"dbeaver-ce-5.0.5-macos.dmg"

I dragged the DBeaver icon into the Applications folder, which as an
administrator did not require any further authentication. My intention
in installing as admin and putting DBeaver into the systemwide
Applications folder was to allow all users to employ DBeaver.

Still as administrator, I started DBeaver, and all went well. It
started, could connect to my running MariaDB, etc. Looks good.

Next, I logged out of the admin account and logged in to my regular
user account. I opened the systemwide Applications folder and found
the DBeaver icon now covered with a circle/slash icon. When I tried to
start DBeaver as my regular user with a doubleclick, I go this popup
message:

  "You can’t open the application “DBeaver” because it is not
  supported on this type of Mac."

Strange! I had just run it from the admin account so it clearly does
work "on this type of Mac."

To check, I simply renamed the DBeaver app to "DBeaver-orig" in the
systemwide Applications folder (using admin authentication).

Next, still logged in as regular user, I opened the DBeaver .dmg file
and again dragged the DBeaver app to the systemwide Applications
folder.  This required authentication as an Administrator, so I
entered the credentials of my admin account and the installation
worked. I how had the newly-installed DBeaver app and the
previously-installed/renamed DBeaver-orig app in the systemwide
Applications folder. From this regular-user account the older
DBeaver-orig app still has the circle/slash icon, but the new
DBeaver.app is clear of it.

<img width="722" alt="dbeaver icons" src="https://user-images.githubusercontent.com/18123804/40151749-55e5e55e-5947-11e8-980e-81072702aeec.png">

Still as regular user, I tried starting the newer DBeaver in the
systemwide Applications directory with a double-click.  This worked
fine, I could connect to MariaDB, etc. atc, as above. So I CAN use
DBeaver as a regular user, but only when installed by that regular
used (with admin authentication).

Next, I logged out as regular user, and logged in as admin user.
From this admin-user account the older DBeaver-orig app now is clear
of the the circle/slash icon, but the new DBeaver app has the
does-not-work icon. So the situation is reversed.

<img width="1029" alt="dbeaver-applications-from-admin-perspective" src="https://user-images.githubusercontent.com/18123804/40155710-d53a6e20-595a-11e8-90c9-79460836e3b4.png">

I tried starting the newer DBeaver app by doubleclicking as admin
user, and I got the same error message as above:

  "You can’t open the application “DBeaver-orig” because it is not
  supported on this type of Mac."

<img width="406" alt="cant-open-popup" src="https://user-images.githubusercontent.com/18123804/40155325-be6e2a94-5958-11e8-991a-5abd52d45394.png">

So there is some kind of asymmetry across users and systemwide
installation procedures for the DBeaver .dmg installer.  I don't see
how to install one copy of DBeaver in the systemwide Applications folder so
that all users (in this case, me-as-admin and regular-me) can start it
in the usual manner. I don't know what leads to the High Sierra
message "...not supported on this type of Mac" but that must be due to
some specific installation factor of apps that indicates they are
supported, that DBeaver has treated anomalously - a
~/Library/ApplicationSupport entry? Some other ~/.dbeaver file? A
missing shell variable? Some /etc file entry?

I'm surprised this hasn't been reported before because it's a
deterministic issue. Use the steps above to reproduce.

Thanks,

- Les

**Comment by mratsim** [comment]
Also impacted, I tried to install dbeaver through 

`brew cask install dbeaver-community`

**Comment by serge-rider** [comment]
I'm not yet sure how to fix this.

Standard workaround is to run `sudo chmod a+x DBeaver.app/Contents/MacOS/*` in Applications folder. This gives execute permissions for all users.

**Comment by gasserles** [comment]
On my machine the problem was that "/Applications/DBeaver.app/Contents/MacOS/dbeaver"  got installed with permissions 700 (rwx ___ ___).  The owner was whoever installed it. So if I installed as regular-me, only I could start it. I changed the permissions of this file (only) to 711 (rwx __x __x) as serge-rider pointed out, and it now works fine for all. 

I don't know where homebrew installs dbeaver, so there might be a different pathname for other installations.

Best, - Les
